src/libraries/Common/src/System/Security/Cryptography/SlhDsa.cs

```diff
@@ -1733,6 +1978,49 @@ private static SlhDsaAlgorithm GetAlgorithmIdentifier(ref readonly AlgorithmIden
             return algorithm;
         }
 
+        private static void ValidateHashAlgorithm(ReadOnlySpan<byte> hash, ReadOnlySpan<char> hashAlgorithmOid)
+        {
+            int? outputSize = hashAlgorithmOid switch
+            {
+                Oids.Md5 => 128 / 8,
+                Oids.Sha1 => 160 / 8,
+                Oids.Sha256 => 256 / 8,
+                Oids.Sha384 => 384 / 8,
+                Oids.Sha512 => 512 / 8,
+                Oids.Sha3_256 => 256 / 8,
+                Oids.Sha3_384 => 384 / 8,
+                Oids.Sha3_512 => 512 / 8,
+                Oids.Shake128 => 256 / 8,
+                Oids.Shake256 => 512 / 8,
+                _ => null,
+            };
+
+            if (outputSize is not null)
+            {
+                if (hash.Length != outputSize)
+                {
+                    throw new CryptographicException(SR.Cryptography_HashLengthMismatch);
```

Comment:
I'm not fully convinced whether we should do this part here. We're straddling the line between:

1) Allow any well-formed OID (no validation on the hash itself)
2) Only allow known OIDs and validate the hash length

Here we allow and well-formed OID *and* validate known hash algos. OSSL does (1) internally while Windows will probably do (2) internally.

Reply:
I feel like it's a reasonable starting place... or we can go all the way to (2) for now.

Either way, when Windows implementations come online, we can make a better informed decision.  It's easier to go from throwing to not than the other way around.